hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java

```diff
@@ -2130,7 +2135,30 @@ public ExchangeRates midnightRates() {
 		return state.tokenAssociations();
 	}
 
-	public FCInvertibleHashMap<MerkleUniqueTokenId, MerkleUniqueToken, OwnerIdentifier> uniqueTokens() {
+	public void rebuildOwnershipsAndAssociations() {
```

Comment:
Lets do the rebuild inside the `TypedTokenStore`. We will need to create new instances of the FCOTMRs since we do not have clear.